perun/postprocess/regressogram/methods.py

```diff
@@ -0,0 +1,135 @@
+"""
+Module with regressogram computational method and auxiliary methods at executing of this method.
+"""
+
+import numpy as np
+import numpy.lib.function_base as np_bin_selectors
+from click import BadParameter
+from scipy import stats
+from sklearn import metrics
+
+from perun.postprocess.regression_analysis.tools import validate_dictionary_keys
+
+# required arguments at regressogram post-processor
+_REQUIRED_KEYS = ['bins', 'statistic']
+
+
+def compute(data_gen, configuration):
+    """
+    The regressogram wrapper to execute the analysis on the individual chunks of resources.
+
+    :param iter data_gen: the generator object with collected data (data provider generators)
+    :param dict configuration: the perun and option context
+    :return: list of dict: the computation results
+    """
+    # checking the presence of specific keys in individual methods
+    validate_dictionary_keys(configuration, _REQUIRED_KEYS, [])
+
+    # list of result of the analysis
+    analysis = []
+    for chunk in data_gen:
+        result = regressogram(chunk[0], chunk[1], configuration['statistic'], configuration['bins'])
+        result['uid'] = chunk[2]
+        result['method'] = 'regressogram'
+        # add partial result to the result list - create output dictionaries
+        analysis.append(result)
+    return analysis
+
+
+def regressogram(x_pts, y_pts, statistic, bins):
+    """
+    Compute the regressogram (binning approach) of a set of data.
+
+    We can view regressogram it as:
+        regressogram = regression + histogram
+
+    Regressogram is a piecewise constant regression function estimator. The x-observation
+    (independent variable) is covered by disjoint bins, and the value of a regressogram in
+    a bin is the mean/median of y-values (dependent variable) for the x-values inside that bin.
+
+    If the 'statistic' contains the int, then this number defines the number of bins. If the
+    'statistic' contains the string from the keys of <_BIN_SELECTORS>, then will be use the chosen
+    method to calculate the optimal bin width and consequently the number of bins.
+
+    :param list x_pts: the list of x points coordinates
+    :param list y_pts: the list of x points coordinates
+    :param str statistic: the statistic to compute
+    :param str/int bins: the number of bins to calculate or the name of computational method
```

Comment:
Suggestion: same as far above, I think it would be wiser to talk about e.g. "buckets" instead of "bins"